### PR TITLE
Handle page error (part of #374)

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**
/cc @inikulin @VasilyStrelyaev

### CODE REVIEW

**Comment by VasilyStrelyaev** [comment]
lgtm

File: src/errors/test-run/index.js [hunk 1]
```diff
@@ -66,6 +66,17 @@ export class DomNodeHybridResultError extends TestRunErrorBase {
 }
 
 
+// Page errors
+//--------------------------------------------------------------------
+export class CouldNotOpenPageError extends TestRunErrorBase {
```
Comment (inikulin) [inline src/errors/test-run/index.js]: `PageLoadError`?

File: src/errors/test-run/templates.js [hunk 2]
```diff
@@ -39,6 +39,12 @@ export default {
         ${err.getCallsiteMarkup()}
     `),
 
+    [TYPE.couldNotOpenPage]: err => markup(err, `
+        Could not open a webpage because of the following error: <code>${err.errMsg}</code>
```
Comment (inikulin) [inline src/errors/test-run/templates.js]: Error message shouldn't be wrapped in `<code>` (see [hammerhead errors](https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/messages.js)). And I think we don't need any prefix - we can pass hammerhead error as is.

```

File: test/functional/fixtures/page-error/testcafe-fixtures/page-error-test.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    await t.click('#unreachable-page-link');
+});
+
+// TODO: action error recovery (GH-567)
+test('Handle', async t => {
+    try {
+        await t.click('#unreachable-page-link');
+    }
+    catch (err) {
+        await t.navigateTo('handled.html');
```
Comment (inikulin) [inline test/functional/fixtures/page-error/testcafe-fixtures/page-error-test.js]: Why we need this error recovery here?

**Comment by inikulin** [comment]
\r-

File: test/functional/fixtures/page-error/testcafe-fixtures/page-error-test.js (hunk 3 above)
Comment (AlexanderMoskovkin) [inline test/functional/fixtures/page-error/testcafe-fixtures/page-error-test.js]: This is scenario when we handle pageError and resume the test from the `about:error` page. But we this scenario will work when we fix #567. Do we need have this test here now?

**Comment by AlexanderMoskovkin** [comment]
FPR

File: test/functional/fixtures/page-error/testcafe-fixtures/page-error-test.js (hunk 3 above)
Comment (inikulin) [inline test/functional/fixtures/page-error/testcafe-fixtures/page-error-test.js]: Nope, doesn't make much sense until we will not get error recovery which should be tested separately

Comment (AlexanderMoskovkin) [inline test/functional/fixtures/page-error/testcafe-fixtures/page-error-test.js]: ok.

btw, FRP

**Comment by inikulin** [comment]
lgtm

**Pull request merged**